### PR TITLE
[sonnen] Fix channel types, Energy should be Power

### LINKED ISSUE #15365: [sonnen] batteryFeedIn, batteryCharging, ... description in Watt is not matching type Number:Energy
batteryFeedIn, batteryCharging, batteryDischarging, consumption, gridFeedIn, gridConsumption are providing Power units expressed in Watt. However the binding sets them up as Number:Energy. 

## Expected Behavior
Provide power units as type Number:Power

## Current Behavior
Power units are provided as Number:Energy. Documentation is also wrong.

### PR DESCRIPTION
**Pull request by lochmueller**
Also adapt the documentation and drop the batteryFeedIn in the documentation, which is not part of the binding.

Fix #15365 / https://github.com/openhab/openhab-addons/issues/15365

The binding define different things/channels in "Number:Energy" but these are regular "W" entries in the binding. So "Number:Power" is the right channel type. This fix should avoid the warnings in the log, that the binding try to put a "W" into the "Number:Energy" item: `Failed to update item 'Sonnen_Battery_Consumption' because '721 W' could not be converted to the item unit 'kWh'`

### CODE REVIEW

**Review by chingon007 — APPROVED** [review]
Ok, looks fine

**Comment by lochmueller** [comment]
I added the "Signed-off-by"-thing to my commit. Sorry for the second round... is my first contribution to openHAB.

**Review by lsiepel — COMMENTED** [review]
LGTM

Only one comment, an upgrade instruction.xml could be added
https://www.openhab.org/docs/developer/bindings/thing-xml.html#updating-thing-types

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks for the fix!

As @lsiepel mentioned, update instructions are needed to fix the channel item types for managed things.

See the linked documentation, and perhaps a similar example in #15002.

**Comment by lochmueller** [comment]
I added the instructions. I hope "targetVersion" => 2 is the right one? Regards, Tim

**Comment by jlaur** [comment]
> I added the instructions. I hope "targetVersion" => 2 is the right one?

It seems right, but you also need to update it here:
https://github.com/openhab/openhab-addons/blob/3b9b0236b46436ce0d91bf95b153eb300f2f3f47/bundles/org.openhab.binding.sonnen/src/main/resources/OH-INF/thing/thing-types.xml#L35

It would be really nice if you could give it a test run in order to avoid regressions. You would need to create a managed Thing with the version prior to your changes. Then uninstall the binding and drop your JAR into the addons directory. The item type for those channels should then automatically change to `Number:Power` for your existing Thing.

**Review by lsiepel — APPROVED** [review]
LGTM
I leave the rest to the addons maintainers.

**Review by chingon007 — APPROVED** [review]

**Comment by jlaur** [comment]
> It would be really nice if you could give it a test run in order to avoid regressions.

Just to be clear: Everything looks fine, I was only waiting for a confirmation of successful migration test. Let me know, and we can merge this PR. 🙂

**Comment by lolodomo** [comment]
@jlaur : I assume noone wants to test the migration ! So maybe you can merge if you think the code looks fine.

**Comment by jlaur** [comment]
> I assume noone wants to test the migration ! So maybe you can merge if you think the code looks fine.

All right, let's give it a wider audience. Fingers crossed. 🙂

**Comment by chingon007** [comment]
I tested it. It works. The change is indeed correct. Please merge

**Pull request merged**
